### ISSUE TITLE
corrupted size vs.prev_size

### ISSUE THREAD

**Issue by Sanchez2020** [dmlc/dgl#3288]
<img width="1201" alt="截屏2021-08-24 上午10 40 19" src="https://user-images.githubusercontent.com/24470627/130547360-f68dff1c-b337-48c1-bf23-fd8675d2c94e.png">

- os：CentOS 7.5
- dgl：0.7.0
- backend：tensorflow 2.5.0
- device：cpu
- python：3.8.3

Is this an installation problem？How to fix it？Thanks to anyone who can help.

**Comment by VoVAllen** [comment]
How did you build/get dgl? What's your python script?

**Comment by Sanchez2020** [comment]
Thank you for your reply.

dgl offline local installation via pip.

python script:

```
import tensorflow as tf
from tensorflow.keras import Model
from dgl.nn.tensorflow import GraphConv


class GCN(Model):
    def __init__(self, in_feats, n_hidden, n_classes):
        super(GCN, self).__init__()
        self.conv1 = GraphConv(in_feats, n_hidden, allow_zero_in_degree=True, activation=tf.nn.rule)
        self.conv2 = GraphConv(n_hidden, n_classes, allow_zero_in_degree=True)

    def call(self, g, h):
        h = self.conv1(g, h)
        h = self.conv2(g, h)
        return h

with tf.device("CPU:0"):
    data = MyDataSet()
    g = data[0]
    feat = g.ndata['feat']
    feat_dim = feat.shape[1]

    model = GCN(in_feats=feat_dim, n_hidden=32, n_classes=2)
    loss_fcn = SparseCategoricalCrossentropy(from_logits=True)
    optimizer = Adam(learning_rate=0.01)

    model(g, feat)    #  This line raise Error

```

**Comment by VoVAllen** [comment]
Hi, what's your cpu model? Because the error message seems related to the AVX instruction introduced recently.

**Comment by Sanchez2020** [comment]
<img width="593" alt="截屏2021-08-24 下午4 14 50" src="https://user-images.githubusercontent.com/24470627/130581901-c78e1102-010d-42a7-8114-67dd89f5b644.png">

How can I avoid this problem？Change dgl the lower version？Which version do you recommend？

**Comment by VoVAllen** [comment]
It's weird and I currently haven't got good idea. Could you try install dgl from source? Following instruction at https://docs.dgl.ai/install/index.html

**Comment by Sanchez2020** [comment]
Thanks for your reply. Please let me know if there is any follow-up progress.

**Comment by VoVAllen** [comment]
I just realized this is probably related to your Dataset. Could you try other dataset such as dataset provided by DGL? Also could you verify whether your edges in your own dataset are right and smaller than the number of nodes?

**Comment by Sanchez2020** [comment]
I checked my custom dataset and found that some nodes are not on the edges. 
Then I modified my data processing code. Now the nodes and edges correspond exactly, but the same error is still reported.

Thank you for your reply anyway.

**Comment by jermainewang** [comment]
Look like the error is from libxsmm. Could you please try building DGL from source with `USE_LIBXSMM=OFF`?

**Comment by Sanchez2020** [comment]
Sorry. My development environment can't connect to the Internet. Thank you for your efforts.

**Comment by sanchit-misra** [comment]
@Sanchez2020: Will it be possible for you to share your code so that we can reproduce at our end?

**Comment by Sanchez2020** [comment]
As mentioned above, the model code and training code are similar. I try to run on a toy graph of a few nodes made by myself, and it's okay. But when I run on a custom dataset with my large data, it raise Error as above. If I have time on Monday, I will share the shape of the dataset and some of the code, but the data can not be shared.

Thank you for your awesome open source project.

**Comment by sanchit-misra** [comment]
@Sanchez2020 : could you try the latest code on master branch? I recently fixed one bug (PR #3345) that may fix this too. 
Also, it would be great if you can share the shape of the dataset. Another way to help us reproduce is to share the graph structure and dummy data for features if that is possible?

**Comment by Sanchez2020** [comment]
The data we used to build the dataset was deleted, so I'm sorry we can't provide the relevant data information. 

I can be sure of a few points：

1. I use graph data built by several points (less than 10) and work normally;
2. I use business data to create graph data (millions of nodes, tens of millions of edges) and run error;
3. I am sure that the shape of the graph data matches, the node id increases continuously from 0, the nodes are all on the edge data, and the nodes on the edge are all on the node data.

I'm sorry, that's all I can provide. Due to the limitation of our time and environment, we can't repeat the error scenario and try the latest version of dgl.  This should be a rare exception. Thank you for your work and reply. When I have the opportunity later, I will try dgl in other projects.
